metis-enrichment/metis-enrichment-client/src/main/java/eu/europeana/enrichment/rest/client/enrichment/RemoteEntityResolver.java

```diff
@@ -0,0 +1,159 @@
+package eu.europeana.enrichment.rest.client.enrichment;
+
+import static eu.europeana.metis.utils.RestEndpoints.ENRICH_ENTITY_EQUIVALENCE;
+import static eu.europeana.metis.utils.RestEndpoints.ENRICH_ENTITY_ID;
+import static eu.europeana.metis.utils.RestEndpoints.ENRICH_ENTITY_SEARCH;
+
+import eu.europeana.enrichment.api.exceptions.UnknownException;
+import eu.europeana.enrichment.api.external.EnrichmentReference;
+import eu.europeana.enrichment.api.external.EnrichmentSearch;
+import eu.europeana.enrichment.api.external.ReferenceValue;
+import eu.europeana.enrichment.api.external.SearchValue;
+import eu.europeana.enrichment.api.external.model.EnrichmentBase;
+import eu.europeana.enrichment.api.external.model.EnrichmentResultBaseWrapper;
+import eu.europeana.enrichment.api.external.model.EnrichmentResultList;
+import eu.europeana.enrichment.api.internal.EntityResolver;
+import eu.europeana.enrichment.api.internal.ReferenceTerm;
+import eu.europeana.enrichment.api.internal.SearchTerm;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
+import org.springframework.web.client.RestClientException;
+import org.springframework.web.client.RestTemplate;
+
+
+public class RemoteEntityResolver implements EntityResolver {
```

Comment:
This class looks good, but some of the lines are too long I think for our coding convention. Perhaps first declare some variables separately (e.g. lambdas) instead of using them inline?

Reply:
I have refactored the code